tools/read-version

```diff
@@ -39,53 +39,62 @@ def which(program):
 
 def is_gitdir(path):
     # Return boolean indicating if path is a git tree.
-    git_meta = os.path.join(path, '.git')
+    git_meta = os.path.join(path, ".git")
     if os.path.isdir(git_meta):
         return True
     if os.path.exists(git_meta):
         # in a git worktree, .git is a file with 'gitdir: x'
         with open(git_meta, "rb") as fp:
-            if b'gitdir:' in fp.read():
+            if b"gitdir:" in fp.read():
                 return True
     return False
 
 
-use_long = '--long' in sys.argv or os.environ.get('CI_RV_LONG')
-use_tags = '--tags' in sys.argv or os.environ.get('CI_RV_TAGS')
-output_json = '--json' in sys.argv
+use_long = "--long" in sys.argv or os.environ.get("CI_RV_LONG")
+use_tags = "--tags" in sys.argv or os.environ.get("CI_RV_TAGS")
+output_json = "--json" in sys.argv
 
 src_version = ci_version.version_string()
 version_long = None
 
 # If we're performing CI for a new release branch (which our tooling creates
 # with an "upstream/" prefix), then we don't want to enforce strict version
 # matching because we know it will fail.
-is_release_branch_ci = (
-    os.environ.get("TRAVIS_PULL_REQUEST_BRANCH", "").startswith("upstream/")
+is_release_branch_ci = bool(
+    os.environ.get("GITHUB_HEAD_REF", "").startswith(f"upstream/{src_version}")
```

Comment:
Should we include both? Will tests running on travis still require the old env var?

Reply:
.travis.yml doesn't invoke any of the tox test targets which leverage this tool even on upstream/XX.YY release branches as seen here in a travis test run https://app.travis-ci.com/github/canonical/cloud-init/builds/254627834
. And as we see on this PR travis succeeded as well even on non-release PRs.

Reply:
Sounds good to me!

Reply:
That said, I'll keep TRAVIS_PULL_REQUEST awareness in this tool in case we grow new tox targets and or want to migrate between Travis or github actions in the future.